### PR TITLE
Expand console template docs.

### PR DESCRIPTION
**Pull request by brian-brazil**
Explain the general structure, prom_query_drilldown and graph library.

### CODE REVIEW

File: content/docs/visualization/consoles.md [hunk 1]
```diff
@@ -25,15 +28,142 @@ Consoles have 5 parts:
...
+The main content is usually graphs. There is a configurable Javascript graphing
+library provided that will handle requesting data from Prometheus, and rendering
 it via [Rickshaw](http://code.shutterstock.com/rickshaw/).
 
 Finally, the table on the right can be used to display statistics in a more
 compact form than graphs.
+
+## Example Console
+
+This is a basic console. It has the number of tasks, how many of them are up,
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: s/has/shows/

File: content/docs/visualization/consoles.md [hunk 2]
```diff
@@ -25,15 +28,142 @@ Consoles have 5 parts:
...
+the average CPU usage, and the average  memory usage in the right hand side
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: right-hand-side graph

queries-per-second graph

(compound adjectives)

Comment (juliusv) [inline content/docs/visualization/consoles.md]: extra space after average?

**Comment by juliusv** [comment]
Will get to the rest of this tomorrow.

**Comment by brian-brazil** [comment]
Thanks, updated for comments thus far.

File: content/docs/visualization/consoles.md [hunk 3]
```diff
@@ -9,13 +9,16 @@ The Console templates allow for creation of arbitrary consoles using the [Go
 templating langauge](http://golang.org/pkg/text/template/). These are served
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: I think I've never reviewed this doc initially. Must have overlooked it somehow. So I'll also point out some existing stuff... like:

s/langauge/language/

Also, in the first sentence, I'd lose the initial "The".

File: content/docs/visualization/consoles.md [hunk 4]
```diff
@@ -9,13 +9,16 @@ The Console templates allow for creation of arbitrary consoles using the [Go
...
 from the Prometheus server.
 
-Console templates are the most powerful way to create templates that can be easily managed in source control, there is a learning curve though so 
-users new to this style of monitoring should try out [PromDash](../promdash/) first.
+Console templates are the most powerful way to create templates that can be
+easily managed in source control, there is a learning curve though so users new
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: s/control, there/control. There/

s/so/, so/

File: content/docs/visualization/consoles.md [hunk 5]
```diff
@@ -9,13 +9,16 @@ The Console templates allow for creation of arbitrary consoles using the [Go
...
+to this style of monitoring should try out [PromDash](../promdash/) first.
 
 ## Getting started
 
-Prometheus comes with an example set of consoles to get you going, these can be found at `/consoles/index.html.example` on a running Prometheus
-and will let you see Node Exporter consoles if your node exporters have a `job="node"` label.
+Prometheus comes with an example set of consoles to get you going, these can be
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: s/going, these/going. These/

File: content/docs/visualization/consoles.md [hunk 6]
```diff
@@ -9,13 +9,16 @@ The Console templates allow for creation of arbitrary consoles using the [Go
...
+found at `/consoles/index.html.example` on a running Prometheus and will
+display Node Exporter consoles if Prometheus is scrapingnode exporters with a
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: s/scrapingnode exporters/scraping Node Exporters/?

Not sure whether we should always capitalize Node Exporter, but as long as we're consistent...

File: content/docs/visualization/consoles.md [hunk 7]
```diff
@@ -9,13 +9,16 @@ The Console templates allow for creation of arbitrary consoles using the [Go
...
+`job="node"` label.
 
 Consoles have 5 parts:
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: This sounds like you are talking about consoles in general, while you are talking about the example templates.

Say "The example consoles have five parts:"

File: content/docs/visualization/consoles.md [hunk 8]
```diff
@@ -25,15 +28,142 @@ Consoles have 5 parts:
 1. The main content in the center, usually graphs
 1. A table on the right
 
-The navigation bar is for links to other systems, such as other Prometheuses, documentation and whatever else makes sense to you.
-The menu is for navigation inside the Prometheus, it's very useful to be able to quickly open a console in another tab to correlate information.
-Both are configured in `console_libraries/menu.lib`.
+The navigation bar is for links to other systems, such as other Prometheuses,
+documentation and whatever else makes sense to you. The menu is for navigation
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: "documentation,"

File: content/docs/visualization/consoles.md [hunk 9]
```diff
@@ -25,15 +28,142 @@ Consoles have 5 parts:
...
+inside the Prometheus, it is very useful to be able to quickly open a console
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: run-on-sentence can be fixed by s/it/which/

"the Prometheus" -> "the same Prometheus server"?

File: content/docs/visualization/consoles.md [hunk 10]
```diff
@@ -25,15 +28,142 @@ Consoles have 5 parts:
...
+in another tab to correlate information. Both are configured in
+`console_libraries/menu.lib`.
 
-The time controls allow changing of the duration and range of the graphs. Console URLs can be shared with others, they'll see the same graphs.
+The time controls allow changing of the duration and range of the graphs.
+Console URLs can be shared with others, they will see the same graphs.
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: "Console URLs can be shared and will show the same graphs for others."?

File: content/docs/visualization/consoles.md [hunk 11]
```diff
@@ -25,15 +28,142 @@ Consoles have 5 parts:
...
 
-The main content is usually graphs. There's a configurable Javascript graphing
-library provided that'll handle requesting data from Prometheus, and rendering
+The main content is usually graphs. There is a configurable Javascript graphing
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: nit: s/Javascript/JavaScript/

File: content/docs/visualization/consoles.md [hunk 12]
```diff
@@ -25,15 +28,142 @@ Consoles have 5 parts:
...
+})
+</script>
+
+{{template "prom_content_tail" .}}
+
+{{template "tail"}}
+```
+
+The `prom_right_table_head` and `prom_right_table_tail` templates contain the
+right hand side table. This is optional.
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: right-hand-side

File: content/docs/visualization/consoles.md [hunk 13]
```diff
@@ -25,15 +28,142 @@ Consoles have 5 parts:
...
+* `humanizeNoSmallPrefix`: For absolute values greater than 1, display the
+  result using [metric prefixes](http://en.wikipedia.org/wiki/Metric_prefix). For
+  absolute values less than 1, display 3 significant digits. This is useful
+  to avoid units such as milliqueries per second that can be produced by
+  `humanize`.
+* `humanize1024`: Display the humanized result using a base of 1024 rather than 1000.
+  This is usually used with `B` as the second argument to produce units such as `KiB` and `MiB`.
+* `printf.3g`: Display 3 significant digits.
+
+Custom formats can be defined, see [prom.lib](https://github.com/prometheus/prometheus/blob/master/console_libraries/prom.lib) for examples.
```
Comment (juliusv) [inline content/docs/visualization/consoles.md]: ". See"

**Comment by brian-brazil** [comment]
I think I've addressed all your comments.

**Pull request merged**